src/System.Runtime.Extensions/tests/System/Environment.GetEnvironmentVariable.cs

```diff
@@ -188,15 +189,40 @@ public void EnumerateYieldsDictionaryEntryFromIEnumerable(EnvironmentVariableTar
         [InlineData(EnvironmentVariableTarget.User)]
         public void EnumerateEnvironmentVariables(EnvironmentVariableTarget target)
         {
-            IDictionary results = Environment.GetEnvironmentVariables(target);
-            foreach (DictionaryEntry result in results)
+            bool lookForSetValue = (target == EnvironmentVariableTarget.Process) || PlatformDetection.IsWindowsAndElevated;
+
+            const string key = "EnumerateEnvironmentVariables";
+            string value = Path.GetRandomFileName();
+
+            try
             {
-                string key = (string)result.Key;
-                string value = (string)result.Value ?? string.Empty;
+                if (lookForSetValue)
+                {
+                    Environment.SetEnvironmentVariable(key, value, target);
+                    Assert.Equal(value, Environment.GetEnvironmentVariable(key, target));
+                }
+
+                IDictionary results = Environment.GetEnvironmentVariables(target);
+
+                // Ensure we can walk through the results
+                IDictionaryEnumerator enumerator = results.GetEnumerator();
+                while (enumerator.MoveNext())
+                {
+                    Assert.NotNull(enumerator.Entry);
+                }
 
-                // Make sure the iterated value we got matches the one we get explicitly
-                Assert.NotNull(result.Key as string);
-                Assert.Equal(value, Environment.GetEnvironmentVariable(key, target));
+                if (lookForSetValue)
+                {
+                    // Ensure that we got our flagged value out
+                    Assert.Equal(value, results[key]);
+                }
+            }
+            finally
+            {
+                if (lookForSetValue)
+                {
+                    Environment.SetEnvironmentVariable(key, null, target);
```

Comment:
This would allow us to still cover the case from the other PR at little cost: ```Yes, and this validation is part of that test, validating that keys returned from the enumeration can be both get and set. IMO it's not superfluous.```